Tests/Pcap++Test/Tests/IpMacTests.cpp

```diff
@@ -355,29 +361,29 @@ PTF_TEST_CASE(TestGetMacAddress)
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "0.255.255.255"), std::invalid_argument, "netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "127.255.255.255"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("invalid"), "255.255.0.0"), std::invalid_argument, "Not a valid IPv4 address");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "invalid"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "999.999.999.999"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "255.255.0.255"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "10.10.10.10"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "0.255.255.255"), std::invalid_argument, "netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(pcpp::IPv4Address("1.1.1.1"), "127.255.255.255"), std::invalid_argument, "netmask is not valid");
 
 	// Invalid c'tor: address + netmask in one string
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("invalid")), std::invalid_argument, "The input should be in the format of <address>/<netmask> or <address>/<prefixLength>");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("invalid/255.255.255.0")), std::invalid_argument, "The input doesn't contain a valid IPv4 network prefix");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/255.255.255.0/24")), std::invalid_argument, "Netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/33")), std::invalid_argument, "Prefix length must be an integer between 0 and 32");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/-1")), std::invalid_argument, "Netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/invalid")), std::invalid_argument, "Netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/999.999.999.999")), std::invalid_argument, "Netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/255.255.0.1")), std::invalid_argument, "Netmask is not valid");
-	PTF_ASSERT_RAISES(pcpp::IPv4Network(std::string("1.1.1.1/0.0.255.255")), std::invalid_argument, "Netmask is not valid");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(std::string("invalid")), std::invalid_argument, "The input should be in the format of <address>/<netmask> or <address>/<prefixLength>");
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4Network(std::string("invalid/255.255.255.0")), std::invalid_argument, "The input doesn't contain a valid IPv4 network prefix: Not a valid IPv4 address");
```

Comment:
Maybe we shouldn't print `e.what()` [here](https://github.com/tigercosmos/PcapPlusPlus/blob/53d1faad0da48ac88a173c8603fd53765577dbfd/Common%2B%2B/src/IpAddress.cpp#L269)? This error message doesn't make a lot of sense, it's enough to keep it `"The input doesn't contain a valid IPv4 network prefix"`

Reply:
we can remove all `e.what()`, but we lose the detail.

 I think it is easier for the users to debug if they see `"The input doesn't contain a valid IPv4 network prefix: Not a valid IPv4 address: 255.255.255.0"` rather than only see `"The input doesn't contain a valid IPv4 network prefix"`

Reply:
But in this case it's pretty clear from the error message that the network prefix is invalid, and the user knows what the input is, so the additional info is redundant

Reply:
I see, let me fix it.

Reply:
fixed.